### ISSUE TITLE
React Compiler warns about functions that carry `"use no memo"`

### ISSUE THREAD

**Issue by bjornpijnacker** [oxc-project/oxc#26140]
A component with a `"use no memo"` directive is not compiled, which is correct, but the compiler warns about it anyway. The warning is the same with and without the directive, so a component that is opted out on purpose keeps printing a warning on every build and every dev start, and there is no way to silence it. `babel-plugin-react-compiler` 1.0.0 prints nothing for the same file.

## Reproduction

`vite.config.ts`

```ts
import react from "@vitejs/plugin-react";
import { defineConfig } from "vite";

export default defineConfig({
    plugins: [react({ compiler: true })],
});
```

`src/Row.tsx`

```tsx
export function Row({ route }: { route: { useSearch: () => string } }) {
    "use no memo";
    const search = route.useSearch();
    return <div>{search}</div>;
}
```

`npm run build`

### Actual

```
vite v8.2.1 building client environment for production...
transforming...[plugin vite:react-compiler] Hooks must be the same function on every render, but this value may change over time

  ! react-compiler(Hooks): Hooks must be the same function on every render,
  | but this value may change over time
   ,-[/src/Row.tsx:3:20]
 2 |     "use no memo";
 3 |     const search = route.useSearch();
   :                    ^^^^^^^|^^^^^^^
   :                           `-- This Hook may change between renders
 4 |     return <div>{search}</div>;
   `----
  help: Call a statically known Hook instead of selecting a Hook dynamically
  note: React Compiler skipped optimizing this component or hook

✓ 2 modules transformed.
```

### Expected

No warning. The component asked not to be compiled, and the output confirms it was not: it comes out unchanged, without memoization.

**Comment by Boshen** [comment]
Thanks for the report. This is the same diagnostic-reporting mismatch fixed by #26128.

React Compiler still validates functions carrying `"use no memo"` and records the bailout as a `CompileError`, but its default `logger: null` keeps recoverable diagnostics silent. Published `oxc-transform-react@0.147.0` predates #26128 and exposed those diagnostics as warnings; current `main` now omits recoverable React Compiler diagnostics at the Node transform boundary while preserving fatal diagnostics.

I verified the exact reproduction against `babel-plugin-react-compiler@1.0.0`, the current local React Compiler source, published `oxc-transform-react`, and Oxc `main`. This should be resolved by the next `oxc-transform-react` release containing #26128.

AI-assisted investigation.
